source/common/common/matchers.cc

```diff
@@ -0,0 +1,102 @@
+#include "common/common/matchers.h"
+
+#include "envoy/api/v2/core/base.pb.h"
+
+#include "common/config/metadata.h"
+#include "common/protobuf/protobuf.h"
+
+#include "absl/strings/match.h"
+
+namespace Envoy {
+namespace Matchers {
+
+bool DoubleMatcher::match(double value) const {
+  switch (matcher_.match_pattern_case()) {
+  case envoy::type::matcher::DoubleMatcher::kRange:
+    return matcher_.range().start() <= value && value < matcher_.range().end();
+  case envoy::type::matcher::DoubleMatcher::kExact:
+    return matcher_.exact() == value;
+  default:
+    return false;
```

Comment:
This is checked by schema, switch to NOT_REACHED. Same elsewhere.

Reply:
Done